nml/version_info.py

```diff
@@ -135,15 +135,23 @@ def get_cli_version():
     return result[0:-1] #strip trailing newline
 
 def get_and_write_version():
+    try:
+        from nml import __version__
+        version = __version__.version
+        if __version__.release:
+            return version
+    except ImportError:
+        pass
     version = get_nml_version()
     if version:
         try:
             path = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
             f = open(os.path.join(path, "nml", "__version__.py"), "w")
             f.write('# this file is autogenerated by setup.py\n')
             f.write('version = "{}"\n'.format(version))
+            f.write('release = {}\n'.format(os.getenv("GITHUB_ACTIONS") != None))
```

Comment:
Clever way to check whether a release is being built. I can imagine that this would not work 100%, (a to-be-written action might try multiple revisions in turn and when someone now manually runs `./setup.py sdist` or other builds, it will not be marked as a release, which might actually be good).

One alternative would be to set `release = True` whenever a build is generated by `setup.py`, but then make sure that it is only set *inside* the generated tarball or wheel, not in the original `nml/__version__.py`. I'm not quite sure how to achieve that, though (other than maybe to simply remove the `nml/__version__.py` automatically after installing it, since you do not really need it in that case).

However, I think that checking for `GITHUB_ACTIONS` will work and is fine for the cases at hand.

Reply:
I very much disagree ;) I think it is a bad idea to use an env-variable that happens to be there during building; the variable might change, or we might change CI, and this is the only part that would depend on that. I am sure there is a better way, as when you need to do these things, often you are solving the wrong problem, or you over-complicated the problem to start with ;)

But mostly, I am missing comments. And lots of them. What is the intention of this code? What did you try to accomplish with reading this env-variable? Please please plllleeeaaaasseeee start adding more comments to your code., especially in functions like this.  This will be so difficult to understand for anyone in the future. Please? :D

Reply:
So what you try to get done is that once a `__version__.py` is generated via a GitHub Actions, it wins from auto-detection. That means this would fail if someone makes a manual release. Strengthen my comment: this sounds like the wrong solution ;)

Why not have a `__version__.py` in the repo that is a dev-stub, and on `build` create the `__version__.py` with the right information? You can even only put that `__version__.py` in the build folder, if I remember correctly, not even changing the repo itself. That means that who-ever makes the source-tarball, or the wheel, it contains the correct and same context? (honest questions, as I am dropping in blind here, trying to discover what has been done :P)